### PR TITLE
DSDEGP-1506 Enable Jenkins to run ITs against deployed Clios

### PR DESCRIPTION
**Pull request by danxmoran**
### Description

These changes let Jenkins run the integration tests against deployed Clio instances without hitting authorization issues. I did a test run of `it:testOnly *EnvIntegrationSpec` [here](https://gotc-jenkins.dsp-techops.broadinstitute.org/job/clio-env-integration-test/3/console). The dev tests all passed, and the staging failures were from our recent renaming of the read-group index, not authorization problems.

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: Ensure that you have added or updated tests
- [x] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

**Review by tbl3rd — APPROVED** [review]

File: clio-client/src/main/scala/org/broadinstitute/clio/client/commands/AddWgsUbamCommand.scala [hunk 1]
```diff
@@ -23,11 +23,16 @@ object AddWgsUbamCommand extends Command {
...
+          bearerToken = config.bearerToken.getOrElse(""),
+          transferWgsUbamV1Metadata = decoded,
+          input = TransferWgsUbamV1Key(
+            flowcellBarcode = config.flowcell.get,
             lane = config.lane.get,
             libraryName = config.libraryName.get,
-            location = Location.pathMatcher(config.location.get)))
+            location = Location.pathMatcher(config.location.get)
+          )
+        )
```
Comment (danxmoran) [inline clio-client/src/main/scala/org/broadinstitute/clio/client/commands/AddWgsUbamCommand.scala:35]: scalafmt doing its thing.

File: clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/EnvIntegrationSpec.scala [hunk 2]
```diff
@@ -14,6 +22,26 @@ abstract class EnvIntegrationSpec(env: String)
     extends BaseIntegrationSpec(s"Clio in $env")
     with IntegrationSuite {
 
+  /** URL of vault server to use when getting bearer tokens for service accounts. */
+  private val vaultUrl = "https://clotho.broadinstitute.org:8200/"
+
+  /**
+    * Path within Vault to the service account info Jenkins should use when talking to Clio.
+    */
+  private val vaultPath = "secret/dsde/gotc/dev/clio/clio-test.json"
```
Comment (tlangs) [inline clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/EnvIntegrationSpec.scala]: Should this be s"secret/dsde/gotc/$env/clio/clio-test.json"?

**Review by tlangs — CHANGES_REQUESTED** [review]

File: clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/EnvIntegrationSpec.scala (hunk 2 above)
Comment (danxmoran) [reply clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/EnvIntegrationSpec.scala]: I'll move it to an env-agnostic path in vault.

Comment (tlangs) [reply clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/EnvIntegrationSpec.scala]: Swole

**Review by tlangs — APPROVED** [review]

**Pull request merged**